tests/chainer_tests/dataset_tests/test_convert.py

```diff
@@ -0,0 +1,197 @@
+import unittest
+
+import numpy
+
+from chainer import cuda
+from chainer import dataset
+from chainer import testing
+from chainer.testing import attr
+
+
+class TestConcatExamples(unittest.TestCase):
```

Comment:
I think we should separate the class and prepare the derived classes for `chack_concat_array`, `check_concat_tuples`, `check_concat_dict` and other things related to `padding` for readability.